### PR TITLE
Allow certificate new

### PR DESCRIPTION
**Pull request by mygnu**
This pr allows to generate `Certificate` manually from strings. useful when user have access to raw data.

### CODE REVIEW

File: src/cert.rs [hunk 1]
```diff
@@ -83,10 +83,12 @@ pub struct Certificate {
 }
 
 impl Certificate {
-    pub(crate) fn new(private_key: String, certificate: String) -> Self {
-        Certificate {
-            private_key,
-            certificate,
+    /// Create Certificate from `String/&str` key and certificate
+    /// useful with manually read files from disk.
+    pub fn new(private_key: impl Into<String>, certificate: impl Into<String>) -> Self {
```
Comment (algesten) [inline src/cert.rs:90]: The doc comment should mention that this is in PEM format to be analogous with for example `private_key` vs `private_key_der`.

**Review by algesten — CHANGES_REQUESTED** [review]
Small doc fix needed.

**Comment by algesten** [comment]
Also notice this have a lot of the same changes as #16, you need to rebase one after the other lands.

File: src/cert.rs (hunk 1 above)
Comment (mygnu) [reply src/cert.rs:90]: Done

**Comment by mygnu** [comment]
Related to `Certificate` struct,  I see that there are a few `unwrap/expect` while parsing with   open ssl. if there is a malformation in the cert or key that may cause panic. panic is something user wouldn't want from a lib. would you consider a pr for returning errors for those methods. we can still unwrap them at the higher level so public api doesn't change..

**Comment by algesten** [comment]
Those `unwrap/expect` comes from the idea that key creation and what is stored in persistence are internal details to acme-lib and thus unless openssl has errors, or the ACME API provider issues incorrect certificates, we should never encounter this.

Opening up this `new` constructor creates a hole in that idea. Not sure what to think about that. The question is why is the user creating a certificate from a PEM and not using the built-in persistence?

**Comment by mygnu** [comment]
I’m working on a project that stores certs and keys in its own ways. and there are potentially thousands of key/cert pairs. It would be very nice to construct from own storage. Even if that’s not the case, we still don’t want the panic if data is corrupted one way or other. As that would take down a potentially large service.

**Pull request closed without merging**